File: App_Resources/Android/src/main/java/com/faceclaw/app/FaceclawCompassListener.java
package com.faceclaw.app;

/** Receives stock compass heading and calibration notifications on the thread that subscribed. */
public interface FaceclawCompassListener {
    void onCompassEvent(int command, int headingDegrees, int magneticAccuracy, int magneticAnomalies,
                        int orientationSource, int diagnosticFlags, long sampleTimeMs);
}

File: native/kotlin/shared/src/commonMain/kotlin/com/faceclaw/app/callbacks/FaceclawCompassListener.kt
package com.faceclaw.app

/** Receives stock compass heading and calibration notifications on the thread that subscribed. */
interface FaceclawCompassListener {
    fun onCompassEvent(
        command: Int,
        headingDegrees: Int,
        magneticAccuracy: Int,
        magneticAnomalies: Int,
        orientationSource: Int,
        diagnosticFlags: Int,
        sampleTimeMs: Long,
    ): Unit
}
